R/stan_module.R

```diff
@@ -0,0 +1,86 @@
+#' StanAll class object
+#' @slot functions Character, the functions part of a stan model
+#' @slot data Character, the data part of a stan model
+#' @slot parameters Character, the parameters part of a stan model
+#' @slot transformed_parameters Character, the transformed_parameters part of a stan model
+#' @slot model Character, the model part of a stan model
+#' @slot prior List, the priors part of a stan model
+#' @slot generated_quantities Character, the generated_quantities part of a stan model
+#' @slot includes Character
+#' @slot inits List with the initial values of the stan model.
+#' @exportClass StanModule
+StanModule <- setClass(
+    "StanModule",
+    representation(
+        functions = "character",
+        data = "character",
+        parameters = "character",
+        transformed_parameters = "character",
+        prior = "list",
+        generated_quantities = "character",
+        inits = "list"
+    )
+)
+
+
+
+#' source_stan_part exports a .stan file as a character
+#' @param filename The name of the .stan file in the directory of the package.
+#' @export
+source_stan_part <- function(filename) {
```

Comment:
I think there needs to be logic here that can handle the scenario of the user actually providing the code directory instead of a file path something like (pseudo code):

```
function (string) {
    if  is_file(string):
        return  read_file(absolute_filename)
    else:
        return string
}
```

Reply:
I use `file.exists` to check whether the string is an existing file and otherwise I use the `system.file` function to locate the file.